src/Cake.Common/Tools/NuGet/Pack/NuspecProcessor.cs

```diff
@@ -24,17 +24,34 @@ public FilePath Process(FilePath nuspecFilePath, NuGetPackSettings settings)
             // Make the nuspec file path absolute.
             nuspecFilePath = nuspecFilePath.MakeAbsolute(_environment);
 
-            // Make sure the nuspec file exist.
             var nuspecFile = _fileSystem.GetFile(nuspecFilePath);
+            XmlDocument xml;
+
+            // Make sure the nuspec file exist.
             if (!nuspecFile.Exists)
             {
-                const string format = "Could not find nuspec file '{0}'.";
-                throw new CakeException(string.Format(CultureInfo.InvariantCulture, format, nuspecFilePath.FullPath));
-            }
+                // Check sufficient parameters supplied voiding the need for nuspec template
+                if (
```

Comment:
Could we break this statement out to a method? Would be easier to read and follow.